### PR TITLE
chore: Update CNS, CNI, NPM Versions and Add Windows Swift CNI Dependency

### PR DESCRIPTION
**Pull request by jaer-tsun**
**What type of PR is this?**

**What this PR does / why we need it**:
Update CNS, CNI, NPM Versions and Add Windows Swift CNI Dependency
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/components.json [hunk 1]
```diff
@@ -83,13 +83,19 @@
...
       ]
     },
     {
       "downloadURL": "mcr.microsoft.com/containernetworking/azure-npm:*",
       "amd64OnlyVersions": [],
-      "multiArchVersions": []
+      "multiArchVersions": [
+        "v1.4.22",
+        "v1.4.27",
+        "v1.4.29"
```
Comment (huntergregory) [inline vhdbuilder/packer/components.json:97]: let's only have 1.4.29 in multiArch, since windows will only be using 29+

Comment (huntergregory) [reply vhdbuilder/packer/components.json:97]: also, might as well not introduce the other two images. 1.4.29 will be the only one in prod soon

Comment (jaer-tsun) [reply vhdbuilder/packer/components.json:94]: updated

**Review by ShiqianTao — APPROVED** [review]

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 2]
```diff
@@ -191,7 +191,9 @@ $global:map = @{
         "https://acs-mirror.azureedge.net/kubernetes/v1.24.0/windowszip/v1.24.0-1int.zip"
     );
     "c:\akse-cache\win-vnet-cni\" = @(
-        "https://acs-mirror.azureedge.net/azure-cni/v1.4.22/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.4.22.zip"
+        "https://acs-mirror.azureedge.net/azure-cni/v1.4.22/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.4.22.zip",
+        "https://acs-mirror.azureedge.net/azure-cni/v1.4.29/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.4.29.zip",
```
Comment (AbelHu) [inline vhdbuilder/packer/generate-windows-vhd-configuration.ps1:195]: it does not exist in sov cloud.
https://mirror.azk8s.cn/azure-cni/azure-cni/v1.4.29/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.4.29.zip

Comment (AbelHu) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:195]: @jaer-tsun how should we use swift cni in Windows nodes? do you plan to update Windows cse scripts?

Comment (jaer-tsun) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:195]: there's a typo in the url you pasted (extra /azure-cni), I'm able to wget

Comment (jaer-tsun) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:195]: we'll be using it for [windows AKS-Swift](https://msazure.visualstudio.com/DefaultCollection/CloudNativeCompute/_git/aks-rp/pullrequest/5914847)

Comment (AbelHu) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:195]: > https://mirror.azk8s.cn/azure-cni/azure-cni/v1.4.29/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.4.29.zip

My mistake. https://mirror.azk8s.cn/azure-cni/v1.4.29/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.4.29.zip works.

Comment (AbelHu) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:195]: > we'll be using it for [windows AKS-Swift](https://msazure.visualstudio.com/DefaultCollection/CloudNativeCompute/_git/aks-rp/pullrequest/5914847)

Will discuss this in the RP PR.

**Review by AbelHu — APPROVED** [review]

**Comment by jaer-tsun** [comment]
moved to #1977

**Pull request closed without merging**